From: Siva Reddy Kallam <siva.kallam@broadcom.com>
Date: Wed, 3 Aug 2016 09:44:00 +0530
Subject: tg3: Report the correct number of RSS queues through tg3_get_rxnfc
Patch-mainline: v4.8-rc3
Git-commit: 9ce6fd7a81e6f787756be2f4b85f4f7bb3658de3
References: bsc#1026030 FATE#321670

This patch remove the wrong substraction from info->data in
tg3_get_rxnfc function. Without this patch, the number of RSS
queues reported is less by one.

Reported-by: Michal Soltys <soltys@ziu.info>
Signed-off-by: Siva Reddy Kallam <siva.kallam@broadcom.com>
Signed-off-by: Michael Chan <michael.chan@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 drivers/net/ethernet/broadcom/tg3.c |    4 ----
 1 file changed, 4 deletions(-)

--- a/drivers/net/ethernet/broadcom/tg3.c
+++ b/drivers/net/ethernet/broadcom/tg3.c
@@ -12552,10 +12552,6 @@ static int tg3_get_rxnfc(struct net_devi
 				info->data = TG3_RSS_MAX_NUM_QS;
 		}
 
-		/* The first interrupt vector only
-		 * handles link interrupts.
-		 */
-		info->data -= 1;
 		return 0;
 
 	default:
